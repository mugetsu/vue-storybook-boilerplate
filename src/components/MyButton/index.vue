<template>
  <button
    class="button"
    :style="{ color, borderColor: color }"
    :class="{ rounded }"
    @click="onClick"
    @dblclick="onDoubleClick"
  >
    <span>{{ label }}</span>
    <!-- @slot This is the default slot -->
    <slot />
  </button>
</template>

<style lang="scss" scoped>
.button {
  span {
    color: blue;
  }
}
</style>

<script>
/**
 * Use `MyButton` to highlight key info with a predefined status.
 */
export default {
  name: "MyButton",
  props: {
    /**
     * Test label
     */
    label: [String, Number],
    /**
     * Test rounded
     */
    rounded: Boolean,
    /**
     * Test color
     */
    color: {
      type: String,
      default: "#42b983"
    }
  },
  methods: {
    onClick($event) {
      /**
       * Emitted when the button is clicked.
       * @event click
       * @type {Event}
       */
      this.$emit("click", $event);
    },
    onDoubleClick($event) {
      /**
       * Emitted when the button is double clicked.
       * @event doubleClick
       * @type {Event}
       */
      this.$emit("dblclick", $event);
    }
  }
};
</script>

<style>
.rounded {
  border-radius: 5px;
}
.button {
  border: 3px solid;
  padding: 10px 20px;
  background-color: white;
  outline: none;
}
</style>
