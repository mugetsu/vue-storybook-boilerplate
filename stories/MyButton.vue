<template>
  <button
    class="button"
    :style="{ color, borderColor: color }"
    :class="{ rounded }"
    @click="onClick"
    @dblclick="onDoubleClick">
    {{ label }}
    <slot/>
  </button>
</template>

<script>
  export default {
    name: 'MyButton',
    props: {
      label: [String, Number],
      rounded: Boolean,
      color: {
        type: String,
        default: '#42b983'
      }
    },
    methods: {
      onClick($event) {
        this.$emit('click', $event);
      },
      onDoubleClick($event) {
        this.$emit('dblclick', $event);
      }
    }
  }
</script>

<style>
  .rounded {
    border-radius: 5px;
  }
  .button {
    border: 3px solid;
    padding: 10px 20px;
    background-color: white;
    outline: none;
  }
</style>
