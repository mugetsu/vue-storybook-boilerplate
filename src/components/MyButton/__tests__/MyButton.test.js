import { shallowMount } from '@vue/test-utils'
import sinon from 'sinon'
import MyButton from '../index.vue'

describe('MyButton', () => {
  let wrapper
  let sandbox

  const generateWrapper = (options) => {
    return shallowMount(MyButton, options)
  }

  beforeEach(() => {
    wrapper = generateWrapper()
    sandbox = sinon.createSandbox()
  })

  afterEach(() => {
    wrapper.destroy()
    sandbox.restore()
  })

  test('is vue', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('render button with label', () => {
    wrapper = generateWrapper({
      propsData: {
        label: 'Label Text'
      }
    })
    expect(wrapper.find('span').text()).toBe('Label Text')
    expect(wrapper.element).toMatchSnapshot()
  })

  test('render button rounded', () => {
    wrapper = generateWrapper({
      propsData: {
        label: 'My Button',
        rounded: true
      }
    })
    expect(wrapper.classes()).toContain('rounded')
    expect(wrapper.element).toMatchSnapshot()
  })

  test('render button with custom color', () => {
    wrapper = generateWrapper({
      propsData: {
        label: 'My Button',
        color: '#FFFFFF'
      }
    })
    expect(wrapper.attributes().style).toBe('color: rgb(255, 255, 255); border-color: #ffffff;')
    expect(wrapper.element).toMatchSnapshot()
  })
})
