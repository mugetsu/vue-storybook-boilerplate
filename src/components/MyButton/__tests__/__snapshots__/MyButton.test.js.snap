// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyButton render button rounded 1`] = `
<button
  class="button rounded"
  style="color: rgb(66, 185, 131); border-color: #42b983;"
>
  <span>
    My Button
  </span>
   
</button>
`;

exports[`MyButton render button with custom color 1`] = `
<button
  class="button"
  style="color: rgb(255, 255, 255); border-color: #ffffff;"
>
  <span>
    My Button
  </span>
   
</button>
`;

exports[`MyButton render button with label 1`] = `
<button
  class="button"
  style="color: rgb(66, 185, 131); border-color: #42b983;"
>
  <span>
    Label Text
  </span>
   
</button>
`;

exports[`MyButton renders correctly 1`] = `
<button
  class="button"
  style="color: rgb(66, 185, 131); border-color: #42b983;"
>
  <span>
    
  </span>
   
</button>
`;
